dll/win32/shell32/CDefView.cpp

```diff
@@ -679,14 +687,27 @@ BOOL CDefView::CreateList()
         dwStyle |= LVS_AUTOARRANGE;
 
     if (m_FolderSettings.fFlags & FWF_SNAPTOGRID)
-        dwExStyle |= LVS_EX_SNAPTOGRID;
-
-    if (m_FolderSettings.fFlags & FWF_DESKTOP)
-        m_FolderSettings.fFlags |= FWF_NOCLIENTEDGE | FWF_NOSCROLL;
+        ListExStyle |= LVS_EX_SNAPTOGRID;
 
     if (m_FolderSettings.fFlags & FWF_SINGLESEL)
         dwStyle |= LVS_SINGLESEL;
 
+    if (m_FolderSettings.fFlags & FWF_FULLROWSELECT)
+        ListExStyle |= LVS_EX_FULLROWSELECT;
+
+    if (m_FolderSettings.fFlags & FWF_SINGLECLICKACTIVATE)
+        ListExStyle |= LVS_EX_TRACKSELECT | LVS_EX_ONECLICKACTIVATE;
+
+    if (m_FolderSettings.fFlags & FWF_NOCOLUMNHEADER)
+        dwStyle |= LVS_NOCOLUMNHEADER;
+
+#if 0
+    // FIXME: Because this is a negative, everyone gets the new flag by default unless they
+    // opt out. This code should be enabled when the shell looks like Vista instead of 2003.
```

Comment:
Maybe with those new versioning helpers this could be done at runtime?

Reply:
In theory, yes, but ROS Explorer still looks like NT5 even when built as NT6 so it will look a little strange with the header visible in all views (IMHO).